islandora.module

```diff
@@ -331,6 +332,50 @@ function islandora_form_alter(&$form, FormStateInterface $form_state, $form_id)
       }
     }
   }
+  $form_object = $form_state->getFormObject();
+  $utils = \Drupal::service('islandora.utils');
+
+  if ($form_object instanceof EntityForm) {
+    $entity = $form_object->getEntity();
+    if ($entity->getEntityTypeId() == "node" && $utils->isIslandoraType($entity->getEntityTypeId(), $entity->bundle()) && strpos($form['#form_id'], 'delete_form') !== FALSE) {
+      $form['delete_associated_content'] = [
+        '#type' => 'checkbox',
+        '#title' => t('Delete all associated medias and nodes'),
+      ];
+
+      $medias = $utils->getMedia($form_state->getFormObject()->getEntity());
+      $media_list = "";
```

Comment:
Thanks for the contribution!

Just a note, rendering markup directly within Drupal is normally frowned upon. That is if anything else later on wanted to modify what was here the markup is already generated and you'd have to resort to hacking on strings.

Alternatively if you utilize the `#theme` key to call `item_list` you can achieve the same thing while still allowing the theme layer/other modules et al to hook in. More information can be found [in the docs](https://www.drupal.org/docs/drupal-apis/render-api/render-arrays#properties) as well as an example can be found [here](https://git.drupalcode.org/project/drupal/-/blob/9.5.x/core/lib/Drupal/Core/Entity/ContentEntityDeleteForm.php#L37-44).